### ISSUE TITLE
Eslint can't recognise vue-property-decorator components and can't lint them

### ISSUE THREAD

**Issue by MohammedSaberMohammed** [typescript-eslint/typescript-eslint#5576]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```typescript
import { Component, Vue } from 'vue-property-decorator';
import { Inject } from 'vue-di-container';
import styles from './verification.module.scss';
import WithRender from './verification.html';
import ServiceA from './ServiceA';
import { VerificationRequest } from '@/models/Verifications';

@WithRender
@Component({
  diProvide: [
  ServiceA,
  ],
  })

export default class Verification extends Vue {
  public styles = styles;
  @Inject(ServiceA)
  public serviceA!: ServiceA;

  public testAPI() {
    this.serviceA.validateFurijatInvoice({} as VerificationRequest);
  }
}
```

### ESLint Config

```javascript
module.exports = {
  root: true,
  env: {
    node: true,
  },

  extends: [
    'plugin:vue/base',
    'plugin:vuetify/base',
    'plugin:vue/essential',
    '@vue/airbnb',
    '@vue/typescript/recommended',
  ],
  parserOptions: {
    ecmaVersion: 2020,
  },
  rules: {
    // TS
    '@typescript-eslint/no-explicit-any': 'off',
    '@typescript-eslint/explicit-module-boundary-types': 'off',
    '@typescript-eslint/no-unused-vars': 'error',
    '@typescript-eslint/no-var-requires': 'off',
    '@typescript-eslint/no-extra-semi': 'off',
    '@typescript-eslint/no-empty-function': 'off',
    '@typescript-eslint/no-this-alias': 'warn',
    // Vue
    'vue/multi-word-component-names': 'off',
    // Other
    camelcase: 'warn',
    'no-underscore-dangle': 'warn',
    'linebreak-style': ['off', 'windows'],
    'consistent-return': 'off',
    indent: ['error', 2, { SwitchCase: 1 }],
    'no-console': 'warn',
    'no-debugger': 'warn',
    'import/no-unresolved': 'off',
    'import/no-cycle': 'warn',
    'import/order': 'off',
    'no-template-curly-in-string': 'off',
    'class-methods-use-this': 'off',
    'lines-between-class-members': 'off',
    'no-return-assign': 'off',
    'max-len': 'off',
    'array-callback-return': 'off',
    'no-shadow': 'off',
    semi: 'error',
    'no-extra-semi': 'off',
    quotes: ['error', 'single'],
    'import/no-extraneous-dependencies': 'off',
    'no-param-reassign': 'off',
    // "import/no-cycle": "",
    'no-plusplus': ['error', { allowForLoopAfterthoughts: true }],
    'import/prefer-default-export': 'off',
    'no-useless-escape': 'off',
    'no-restricted-globals': 'warn',
    'no-extend-native': 'off',
    'import/extensions': 'off',
    'import/no-dynamic-require': 'off',
    'global-require': 'off',
    'unicode-bom': 'off',
    'no-restricted-syntax': 'off',
    'no-prototype-builtins': 'warn',
    'no-useless-concat': 'warn',
    'prefer-destructuring': 'warn',
    'no-mixed-operators': 'warn',
    'no-use-before-define': 'warn',
    'import/no-useless-path-segments': 'error',

  },
  overrides: [
    {
      files: [
        '**/__tests__/*.{j,t}s?(x)',
        '**/tests/unit/**/*.spec.{j,t}s?(x)',
      ],
      env: {
        jest: true,
      },
    },
  ],
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "strict": true,
    "jsx": "preserve",
    "importHelpers": true,
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "forceConsistentCasingInFileNames": true,
    "useDefineForClassFields": true,
    "sourceMap": true,
    "baseUrl": ".",
    "types": [
      "node",
      "webpack-env",
      "jest"
    ],
    "paths": {
      "@/*": [
        "src/*"
      ]
    },
    "lib": [
      "esnext",
      "dom",
      "dom.iterable",
      "scripthost"
    ]
  },
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "tests/**/*.ts",
    "tests/**/*.tsx", 
    "src/shims-html.ts"  
  ],
  "exclude": [
    "node_modules"
  ]
}
```

### Expected Result

Component, Inject and WithRender Should be recognized and working fine also the indentation with decorators should work just fine .

### Actual Result

![image](https://user-images.githubusercontent.com/19964514/187472493-60b5f7ab-2988-4e43-9acc-41fb95e9aeaa.png)
![eslint-issue](https://user-images.githubusercontent.com/19964514/187476696-238fb56c-5579-410f-9771-1c30aa43d28e.gif)

### Additional Info

I'm using vue create and created a template based on ` eslint, vue, and vue-template-loader` and used vue like angular as in [here](https://www.digitalocean.com/community/tutorials/vuejs-vue-template-loader)  and it was working till yesterday.

but today about 4 apps + all piplines failes in build step because of eslint always tells that:
1 - decorators from  vue-property-decorator, vue-di-container and withRender ( are not used even they are used ).
2- linter Indentation got confused when linting decorator

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `^5.4.0` |
| `@typescript-eslint/parser`            | `^5.4.0` |
| `@typescript-eslint/scope-manager`     | `Not-used` |
| `@typescript-eslint/typescript-estree` | `Not-used` |
| `@typescript-eslint/type-utils`        | `Not-used` |
| `@typescript-eslint/utils`             | `Not-used` |
| `TypeScript`                           | `~4.5.5` |
| `ESLint`                               | `^7.32.0` |
| `node`                                 | `14.19.0` |
|`vue-template-compiler`       | `^2.6.14`|
 |`vue-template-loader`  |`^1.1.0`|

**Comment by bradzacher** [comment]
There was a bug that has been fixed in the latest version released not long ago.

Upgrade to the latest version.

**Comment by bradzacher** [comment]
You should always keep the two packages at the same version!
So upgrade both!
